packages/model-viewer/src/features/scene-graph/texture-factory.ts

```diff
@@ -0,0 +1,66 @@
+
+import {MeshStandardMaterial, Texture as ThreeTexture} from 'three';
+
+import {GLTF, Texture as GLTFTexture, TextureInfo as GLTFTextureInfo} from '../../three-components/gltf-instance/gltf-2.0.js';
+
+import {Image} from './image.js';
+import {Sampler} from './sampler.js';
+import {TextureInfo, TextureUsage} from './texture-info.js';
+import {Texture} from './texture.js';
+
+
+
+export const $createTexture = Symbol('createTexture');
+export const $createTextureInfo = Symbol('createTextureInfo');
+
+/**
+ * Helper class for creating Textures and TextureInfos.
+ */
+export class TextureFactory {
+  // Creates context from Texture. This creats a texture with a valid
+  static[$createTexture](threeTexture: ThreeTexture): Texture {
+    // Creates an empty glTF data set, allows for creating a texture that's
+    // not bound to a material.
+    const emptyOnUpdate = () => {};
+    return new Texture(
```

Comment:
creates a texture from a ThreeJS texture, notice the glTF data is just null instead of being spoofed. Much simpler to write, use, and read this code.